src/FormulaParser.cpp

```diff
@@ -0,0 +1,70 @@
+#include "FormulaParser.h"
+#include <iostream>
+#include <fstream>
+
+
+FormulaParser::FormulaParser() {}
+
+FormulaParser &FormulaParser::getInstance() {
+    static FormulaParser instance;
+    if (!instance.isInitialized_) {
+        instance.initialize();
+    }
+    return instance;
+}
+
+void FormulaParser::initialize() {
+    
+    readFormulasFromConfig();
+    isInitialized_ = true;
+}
+
+double FormulaParser::evaluate(const std::string &formulaName, double power, double weight, double speed, double verbrauch, double tank) const {
+
+    auto formulaIter = formulas_.find(formulaName);
+    if (formulaIter == formulas_.end()) 
+    {
+        std::cout << "Formula for " << formulaName << " not found" << std::endl;
```

Comment:
cout is discouraged in this code-base - mainly because it uses it's own "central" logging with AT_Log. Example would be: https://github.com/WizzardMaker/AirlineTycoon/blob/366c0a9bd17dd7c1b77a526b7a874a42f319294b/src/SBLib/source/Network.cpp#L364
Where the first argument is the originator of the message (Here would probably be "FormularParser"), second argument is the formatting string (printf style formatting) and var arg being the to format objects

See here for definition:
https://github.com/WizzardMaker/AirlineTycoon/blob/366c0a9bd17dd7c1b77a526b7a874a42f319294b/src/TeakLibW.h#L32-L41